include/groonga/expr.h

```diff
@@ -134,6 +134,27 @@ grn_expr_append_op(grn_ctx *ctx, grn_obj *expr, grn_operator op, int nargs);
 GRN_API grn_rc
 grn_expr_get_keywords(grn_ctx *ctx, grn_obj *expr, grn_obj *keywords);
 
+/**
+ * \brief Escape target characters in a query by a specified escape character.
+ *
+ * \param ctx The context object. Its encoding must match that of the input
+ *            \p query.
+ * \param query The input query to be escaped.
+ * \param query_size The byte size of \p query. A value of -1 indicates that
+ *                   the query is NULL terminated.
```

Comment:
```suggestion
 *                   the query is null-terminated.
```

or "terminated with \`NUL\`".

In general, `NULL` is used for "null pointer" (`(void *)0`) in C context. `NUL` is used for "null character" (`'\0'`) in C context. Values of both of them are `0` but C string uses "null character" not "null pointer". So "null-terminated" or "terminated with \`NUL\`" is better.

Reply:
fix: [a4218bb](https://github.com/groonga/groonga/pull/2268/commits/a4218bb5bc2f0b6fa84d17d8dfd66d2b4933c23b) Thanks you so much. I understood the difference in C context!